javascript-style-guide.md

```diff
@@ -34,24 +37,67 @@ avoid whitespace-oriented commits and diffs.
 
 Use JSCS to validate the format of your code. Your editor may offer JSCS
 integration, which you may find useful if you are not familiar with the style
-guide. You can also use our projects' Grunt interface.
+guide. You can also use our projects' Grunt or Gulp interfaces.
+
+# English
+
+## Spelling
 
-# Grammar
+Use real words for identifiers.
 
-## Use real words for identifiers.
+For instance, "element" is best written as "element", not "el", "elm", "elem",
+or "e".
 
-"element" is best written as "element", not "el", "elm", "elem", or "e".
+For your convenience, the following words also have a preferred spelling:
+
+Word      | Best written as | Known to be written as
+----------|-----------------|-----------------------
+directory | directory       | dir
+error     | error           | err, e
+index     | index           | idx, i
+request   | request         | req
+response  | response        | res
 
 Ubiquitous acronyms such as "http" and "url" are permissible.
 
-## Capitalize only the first letter of words in identifiers.
+## Capitalization
+
+Capitalize only the first letter of words in identifiers.
+
+For instance, a phrase such as "File Transfer URI", which contains a
+normally-capitalized acronym, has the identifier `fileTransferUri` (as opposed
+to something like `fileTransferURI`).
+
+## Word choice
+
+Prefer descriptiveness, then brevity.
+
+When creating APIs with hooks for certain events, prefix names with "on". e.g.,
+`onComplete`, `onFail`.
 
-A phrase such as "File Transfer URI", which contains a normally-capitalized
-acronym, has the identifier `fileTransferUri` (as opposed to something like
-`fileTransferURI`).
+When talking about millisecond values since the Unix epoch, suffix names with
+"time". e.g., `startTime = Date.now()`.
+
+When naming predicate values or functions, prefix names with "is". e.g.,
+`isReady`, `isAvailable()`.
+
+Functions which return objects shoud be prefixed with "make". e.g.,
+`makeMammal()`, `makeCat()`.
 
 # Syntax
 
+## Variables
+
+If ECMAScript 6 features are available, prefer `const`. Use `let` only when the
+value needs to change.
+
+Thanks to block scope and temporal dead zones, `const` and `let` variables need
+not be defined at the top of a function, nor in single, comma-delimited
+statements. In fact, they should be defined separately, and close to their site
+of first use.
```

Comment:
Okay, I will add a techdebt item. JSCS plans to make a 2.0 release soon, at which time `requireVarDeclFirst` will become available, so we can make the change then.